### PR TITLE
[REVIEW] Add backward compatibility for `dask-cudf` to work with other versions of `dask`

### PR DESCRIPTION
**Pull request by galipremsagar**
The current `dask-cudf` includes latest changes that work with latest `dask` changes. But when an existing release version of `dask` is used with `dask-cudf` it will fail to import some of the dispatches needed for `make_meta` & `make_meta_util`.. This PR will handle those import errors and have a fallback in place.

### CODE REVIEW

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Ah I have those changes in PR ( https://github.com/rapidsai/cudf/pull/8363 )

**Comment by kkraus14** [comment]
> Ah I have those changes in PR ( #8363 )

I think that just bumps the required version whereas this PR allows the older version to continue working. Going to close it in favor of this PR.

File: conda/environments/cudf_dev_cuda11.0.yml [hunk 1]
```diff
@@ -41,8 +41,8 @@ dependencies:
   - mypy=0.782
   - typing_extensions
   - pre_commit
-  - dask==2021.4.0
-  - distributed>=2.22.0,<=2021.4.0
+  - dask==2021.5.0
```
Comment (kkraus14) [inline conda/environments/cudf_dev_cuda11.0.yml:44]: Can these and all the other ones be `>=2021.4.0,<=2021.5.0`?

Comment (galipremsagar) [reply conda/environments/cudf_dev_cuda11.0.yml]: Done

**Comment by kkraus14** [comment]
I think this needs to be updated as well: https://github.com/rapidsai/cudf/blob/branch-21.06/conda/recipes/custreamz/meta.yaml#L34-L35

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
@gpucibot merge

**Review by raydouglass — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 😄

**Pull request merged**